tests/test_recreation.py

```diff
@@ -1187,6 +1187,51 @@ def test_bad_predictor_table_header(self):
             actual_messages.add(error_strings)
         self.assertTrue(expected_message in actual_messages)
 
+    def test_validate_predictor_types(self):
+        """Recreation Validate: assert error on invalid type value"""
+        from natcap.invest.recreation import recmodel_client
+
+        predictor_id = 'dem90m'
+        # include trailing whitespace in the type, this should pass
+        raster_path = os.path.join(SAMPLE_DATA, 'predictors/dem90m_coarse.tif')
+        table_path = os.path.join(self.workspace_dir, 'table.csv')
+        with open(table_path, 'w') as file:
+            file.write('id,path,type\n')
+            file.write(f'{predictor_id},{raster_path},raster_mean \n')
+
+        # include a typo in the type, this should fail
+        bad_table_path = os.path.join(self.workspace_dir, 'bad_table.csv')
+        with open(bad_table_path, 'w') as file:
+            file.write('id,path,type\n')
+            file.write(f'{predictor_id},{raster_path},raster?mean\n')
+
+        args = {
+            'aoi_path': os.path.join(SAMPLE_DATA, 'andros_aoi.shp'),
+            'cell_size': 40000.0,
+            'compute_regression': True,
+            'start_year': '2005',
+            'end_year': '2014',
+            'grid_aoi': False,
+            'predictor_table_path': table_path,
+            'workspace_dir': self.workspace_dir,
+        }
+
+        # there should be no error when the type has trailing whitespace
+        recmodel_client.execute(args)
+        output_path = os.path.join(self.workspace_dir, 'regression_coefficients.txt')
+        print('regression_coefficients.txt:')
+
+        # the regression_coefficients.txt output file should contain the
+        # predictor id, meaning it wasn't dropped from the regression
+        with open(output_path, 'r') as output_file:
+            self.assertTrue(predictor_id in ''.join(output_file.readlines()))
+
+        # try it again with the table that has a misspelled type
+        # it should raise an error
+        args['predictor_table_path'] = bad_table_path
+        with self.assertRaises(ValueError):
+            recmodel_client.execute(args)
+

```

Comment:
Could I suggest breaking this into two tests? One that tests for predictor id being used with whitespace and one that fails from typo? I think it's best to have each test in charge of one thing.

Also for the `assertRaises` test, could test against the error message? I feel like it's prudent to make sure we are not just catching an arbitrary `ValueError` that could have come from somewhere else.
```python
with self.assertRaises(ValueError) as cm:
    recmodel_client.execute(args)
self.assertTrue("Expected error message chunk here" in str(cm.exception))
```

Thanks!